java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -2845,21 +3122,181 @@ protected boolean cleanImpl(boolean logErrorIfNotClean) {
+        mainValidDevBuff.copyFromHostBuffer(mainColValid, 0, validLen);
+      }
+      if (data != null) {
+        long dataLen = ColumnVector.getNativeValidPointerSize(mainColRows);
+        mainDataDevBuff = DeviceMemoryBuffer.allocate(dataLen);
+        mainDataDevBuff.copyFromHostBuffer(data, 0, dataLen);
+      }
+      if (mainColOffsets != null) {
+        long offsetsLen = OFFSET_SIZE * (mainColRows + 1);
+        mainOffsetsDevBuff = DeviceMemoryBuffer.allocate(offsetsLen);
+        mainOffsetsDevBuff.copyFromHostBuffer(mainColOffsets, 0, offsetsLen);
+      }
+      return new ColumnVector(mainColType, mainColRows, nullCount, mainDataDevBuff, mainValidDevBuff, mainOffsetsDevBuff, devChild);
+    }
+
+    private static NestedColumnVector createNewNestedColumnVector(HostColumnVector.NestedHostColumnVector nestedChildren) {
+
+      if (nestedChildren == null) {
+        return null;
+      }
+      DType colType = nestedChildren.getType();
+      Optional<Long> nullCount = nestedChildren.getNullCount();
+      //TODO check long to int is ok
+      int colRows = (int) nestedChildren.getRows();
+      HostMemoryBuffer colData = nestedChildren.getNestedChildren().isEmpty() ? nestedChildren.getData() : null;
+      HostMemoryBuffer colValid = nestedChildren.getValidity();
+      HostMemoryBuffer colOffsets = nestedChildren.getOffsets();
+
+      // assuming list for now
+      return createNestedColumnVector(colType, colRows, nullCount, colData, colValid, colOffsets,
+          createNewNestedColumnVector(nestedChildren.getNestedChildren().isEmpty() ? null : nestedChildren.getNestedChildren().get(0)));
+    }
+
+    private long getViewHandle() {
+      long childColumnViewHandle = 0l;
+      if (child != null) {
+        childColumnViewHandle = child.getViewHandle();
+      }
+      long dataAddr = data == null ? 0 : data.address;
+      long dataLen = data == null ? 0 : data.length;
+      long offsetAddr = offsets == null ? 0 : offsets.address;
+      long validAddr = valid == null ? 0 : valid.address;
+      int nc = nullCount.orElse(OffHeapState.UNKNOWN_NULL_COUNT).intValue();
+      return makeCudfColumnView(dataType.nativeId, dataAddr, dataLen, offsetAddr, validAddr, nc, rows, childColumnViewHandle);
+    }
+
+    private List<DeviceMemoryBuffer> getBuffersToClose() {
+      List<DeviceMemoryBuffer> buffers = new ArrayList<>();
+      if (child != null) {
+        buffers.addAll(child.getBuffersToClose());
+      }
+      if (data != null) {
+        buffers.add(data);
+      }
+      if (valid != null) {
+        buffers.add(valid);
+      }
+      if (offsets != null) {
+        buffers.add(offsets);
+      }
+      return buffers;
+    }
+
+    private static long getEndStringOffset(long index, HostMemoryBuffer offsets) {
+      //TODO check for row count wrt to index
+      return offsets.getInt((index + 1) * 4);
+    }
+
+    static void printBuffer(HostMemoryBuffer buffer) {
```

Comment:
If we want this for debugging then we should put it in HostMemoryBuffer instead.  Or else we should delete it.